File: VbNetApp/BusinessLogic/Commands/AutoCADExceptions.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports Autodesk.AutoCAD.ApplicationServices

Namespace Commands
    Public Class AutoCADExceptions

        Public Shared Sub Handle(ex As Exception)

            Dim doc As Document = Application.DocumentManager.MdiActiveDocument
            Dim stackTraceStringList As List(Of String) = cadwiki.NetUtils.Exceptions.GetStackTraceLines(ex)
            Dim stackTraceString As String = cadwiki.NetUtils.Lists.StringListToString(stackTraceStringList, vbLf)
            For Each str As String In stackTraceStringList
                doc.Editor.WriteMessage(vbLf & str)
            Next
            doc.Editor.WriteMessage(vbLf & "Exception :" & ex.Message)
            doc.Editor.WriteMessage(vbLf & vbLf)
        End Sub

    End Class

End Namespace

File: CSharpApp/BusinessLogic/Commands/AutoCADExceptions.cs
using System;
using Microsoft.VisualBasic;

namespace BusinessLogic.Commands
{
    public class AutoCADExceptions
    {

        public static void Handle(Exception ex)
        {

            var doc = Autodesk.AutoCAD.ApplicationServices.Core.Application.DocumentManager.MdiActiveDocument;
            var stackTraceStringList = cadwiki.NetUtils.Exceptions.GetStackTraceLines(ex);
            string stackTraceString = cadwiki.NetUtils.Lists.StringListToString(stackTraceStringList, Environment.NewLine);
            foreach (string str in stackTraceStringList)
                doc.Editor.WriteMessage(Environment.NewLine + str);
            doc.Editor.WriteMessage(Environment.NewLine + "Exception :" + ex.Message);
            doc.Editor.WriteMessage(Environment.NewLine + Environment.NewLine);
        }

    }

}